source/slang/slang-diagnostic-defs.h

```diff
@@ -744,6 +744,10 @@ DIAGNOSTIC(45001, Error, unresolvedSymbol, "unresolved external symbol '$0'.")
 DIAGNOSTIC(41201, Warning, expectDynamicUniformArgument, "argument for '$0' might not be a dynamic uniform, use `asDynamicUniform()` to silence this warning.")
 DIAGNOSTIC(41201, Warning, expectDynamicUniformValue, "value stored at this location must be dynamic uniform, use `asDynamicUniform()` to silence this warning.")
 
+
+DIAGNOSTIC(41202, Error, notEqualBitCastSize, "trying to bit_cast not equally sized type '$0' with size '$1' into type '$2' with size '$3'")
+DIAGNOSTIC(41203, Warning, notEqualReinterpretCastSize, "trying to reinterpret not equally sized type '$0' with size '$1' into type '$2' with size '$3'")
```

Comment:
Feel free to ignore this but I would prefer message more like,
```
Invalid reinterpret_cast when the sizes of given types are different
```

Reply:
I will adjust the wording to be clearer. 
I think I will still include the type+size_of part of the warning. This is an easy to make (but hard to find) mistake. 
I will use `reinterpret<>` instead of reinterpret_cast since Slang code uses `reinterpret<>` syntax.